root/etc/e-smith/templates/etc/httpd/conf.d/default-virtualhost.inc/50cgp

```diff
@@ -0,0 +1,26 @@
+#
+# 50cgp
+#
+{
+    use esmith::ConfigDB;
+    use esmith::util;
+
+    my $alias = $cgp{'alias'} || "";
+
+    # initialize alias if needed
+    if ($alias eq "") {
+        $alias = esmith::util::genRandomHash();
+        my $confdb = esmith::ConfigDB->open();
+        $confdb->set_prop('cgp','alias',$alias);
```

Comment:
I don't like a `set_prop()` in a template, however that's the way it works since ages...

Reply:
I agree with you. but I'd tried to minimize changes :(